ci/env.sh

```diff
@@ -5,39 +5,40 @@
 set -euxo pipefail
 
 NAME=nordvpn
-export NAME
-
 HASH=$(git rev-parse --short HEAD)
-export HASH
 
 VERSION_PATTERN="^[0-9]+\.[0-9]+\.[0-9]+$"
 if [[ "${CI_COMMIT_TAG:-}" =~ ${VERSION_PATTERN} ]]; then
   ENVIRONMENT="prod"
-  export ENVIRONMENT
-
   REVISION=1
-  export REVISION
-
   VERSION="${CI_COMMIT_TAG}"
-  export VERSION
 
-  # version date should be always the same
-  VERSION_DATE="$(git log -1 --format=%aI "${CI_COMMIT_TAG}" | xargs -I{} date -u -d "{}" +"%Y-%m-%dT%H:%M:%SZ")"
-  export VERSION_DATE
+  # version date should be always the same;
+  # check if the tag actually exists in git, else use HEAD's commit date
+  if git rev-parse "${CI_COMMIT_TAG}" >/dev/null 2>&1; then
+    VERSION_DATE="$(git log -1 --format=%aI "${CI_COMMIT_TAG}" | xargs -I{} date -u -d "{}" +"%Y-%m-%dT%H:%M:%SZ")"
+  else
+    echo "Warning: Git tag '${CI_COMMIT_TAG}' not found, using HEAD instead"
+    VERSION_DATE="$(git log -1 --format=%aI HEAD | xargs -I{} date -u -d "{}" +"%Y-%m-%dT%H:%M:%SZ")"
+  fi
```

Comment:
Can this get extracted into a separate method, and reused here, and below (so we won't have this much code repetition)?